File: services/api-principal/src/main/java/com/guminteligencia/ura_chatbot_ia/infrastructure/repository/entity/UsuarioEntity.java
package com.guminteligencia.ura_chatbot_ia.infrastructure.repository.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.util.Map;
import java.util.UUID;

@Entity(name = "Usuario")
@Table(name = "usuarios")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
public class UsuarioEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    private String nome;
    private String telefone;
    private String senha;
    private String email;

    @Column(name = "telefone_conectado")
    private String telefoneConectado;

    @Column(name = "atributos_qualificacao")
    @JdbcTypeCode(SqlTypes.JSON)
    private Map<String, Object> atributosQualificacao;

    @Column(name = "configuracao_crm")
    @Embedded
    private ConfiguracaoCrmEntity configuracaoCrm;

    @Column(name = "mensagem_direcionamento_vendedor")
    private String mensagemDirecionamentoVendedor;

    @Column(name = "mensagem_recontato_g1")
    private String mensagemRecontatoG1;

    @Column(name = "mensagem_encaminhamento")
    private String mensagemEncaminhamento;

    @Column(name = "whatsapp_token")
    private String whatsappToken;

    @Column(name = "whatsapp_id_instance")
    private String whatsappIdInstance;

    @Column(name = "agente_api_key", columnDefinition = "TEXT")
    private String agenteApiKey;
}

File: services/api-agente/src/infrastructure/entity/usuario_entity.py
from sqlalchemy import Column, BINARY, JSON, String

from src.config.base_class import Base


class UsuarioEntity(Base):

    __tablename__ = "usuarios"

    id = Column(BINARY(16), primary_key=True, index=True)
    nome = Column(String(255), nullable=False)
    telefone = Column(String(20), nullable=False)
    senha = Column(String(255), nullable=False)
    email = Column(String(255), nullable=False)
    telefone_conectado = Column(String(20), nullable=False)
    atributos_qualificacao = Column(JSON, nullable=False)
    agente_api_key = Column(String(255), nullable=False)